### PR TITLE
DHFPROD-1924: Metadata and options recording (add option.headers)

### PR DESCRIPTION
**Pull request by hao1st**
https://project.marklogic.com/jira/browse/DHFPROD-1924

### CODE REVIEW

File: marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/builtins/steps/ingest/default/main.sjs [hunk 1]
```diff
@@ -27,13 +23,25 @@ function main(id, content, options) {
...
 
   return envelope;
 }
 
+function createHeaders(id, content, options) {
+  let headers = {};
+  for (let key in options.headers) {
+    headers[key] = options.headers[key];
+    if (headers[key] == datahub.flow.consts.CURRENT_DATE_TIME) {
+      headers[key] = fn.currentDateTime;
```
Comment (aebadirad) [inline marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/builtins/steps/ingest/default/main.sjs]: should be fn.currentDateTime()

File: marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/builtins/steps/ingest/default/main.sjs [hunk 2]
```diff
@@ -27,13 +23,25 @@ function main(id, content, options) {
...
-
+  let headers = createHeaders(id, content, options);
   let envelope = datahub.flow.flowUtils.makeEnvelope(instance, headers, triples, outputFormat);
 
   return envelope;
 }
 
+function createHeaders(id, content, options) {
+  let headers = {};
+  for (let key in options.headers) {
```
Comment (aebadirad) [inline marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/builtins/steps/ingest/default/main.sjs]: Is there a way we can functionize this to the hubUtils library for checking over the constants and returning the value so we just need to loop over the object and use the function to check and return the correct value?

File: marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/builtins/steps/ingest/default/main.sjs (hunk 1 above)
Comment (hao1st) [reply marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/builtins/steps/ingest/default/main.sjs]: yes. i missed it when I copied from qconsole.

File: marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/builtins/steps/ingest/default/main.sjs (hunk 2 above)
Comment (hao1st) [reply marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/builtins/steps/ingest/default/main.sjs]: Sure. we can do that.

**Pull request merged**